### ISSUE TITLE
ImageSimulationEngine oversampling breaks Torch autograd

### ISSUE THREAD

**Issue by somnus-J-307** [optiland/optiland#720]
### Checklist

- [x] I have searched the existing issues and discussions for a similar problem.
- [x] I have read the image-simulation documentation and gallery example and did not find this limitation documented.
- [x] I am using the latest Optiland master branch.
- [x] I have reproduced and debugged the issue.
- [x] I have included the relevant environment details and a minimal reproducer.

### Bug Report

**Describe the bug**

When the Torch backend is active, `ImageSimulationEngine` silently disconnects the source image from the autograd graph whenever `oversample > 1`.

There are two causes on the current `master` branch:

1. `_preprocess()` converts the Torch tensor to NumPy for `scipy.ndimage.zoom`, then creates a new backend array. The new tensor is not connected to the input tensor.
2. `_postprocess()` checks `be.get_backend().__class__.__name__ == "TorchBackend"`, but `be.get_backend()` returns the string `"torch"`. The intended differentiable Torch branch is therefore unreachable, and postprocessing also goes through NumPy.

This makes the oversampling option incompatible with differentiable image-simulation or training workflows.

**To reproduce**

    import torch

    import optiland.backend as be
    from optiland.analysis.image_simulation import ImageSimulationEngine

    be.set_backend("torch")
    be.grad_mode.enable()

    engine = ImageSimulationEngine(
        optic=None,
        config={"oversample": 2, "padding": 1},
    )
    image = torch.ones(
        (1, 1, 4, 5),
        dtype=torch.float64,
        requires_grad=True,
    )
    engine.source_image = image

    upsampled, pad_info = engine._preprocess(image)
    result = engine._postprocess(upsampled, pad_info)

    print(upsampled.grad_fn)  # None
    print(torch.autograd.grad(result.sum(), image, allow_unused=True)[0])  # None

**Expected behavior**

With the Torch backend, oversampling should:

- remain connected to the source-image autograd graph;
- preserve dtype and device;
- preserve the existing NumPy/SciPy sampling geometry;
- return the same expected shapes after upsampling, downsampling, and cropping.

**Environment**

- Optiland Version: `0.6.1`, current `master` at `bc9566a51b43b0acf63dfd2726a06c59c24a2913`
- Python Version: `3.13.13`
- PyTorch Version: `2.13.0+cpu`
- OS: Windows

**Proposed contribution**

I would like to submit a focused PR that:

- keeps the current SciPy path unchanged for the NumPy backend;
- uses `torch.nn.functional.interpolate` for Torch preprocessing and postprocessing;
- fixes the backend-name check;
- uses endpoint-aligned interpolation to match the existing `scipy.ndimage.zoom` geometry;
- adds NumPy/Torch regression coverage for numerical parity, shape, dtype, device, and non-empty finite gradients.

The change would be limited to the image-simulation engine and its tests, with no new dependency. I have a local patch and targeted tests ready and would be happy to open the PR if this approach is acceptable.

**Screenshots**

Not applicable.
